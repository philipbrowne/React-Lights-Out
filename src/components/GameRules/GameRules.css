.GameRules {
  color: orange;
  display: flex;
  justify-content: center;
  width: 80%;
  text-align: center;
}

@media (max-width: 600px) {
  .GameRules {
    font-size: 0.8rem;
  }
}
