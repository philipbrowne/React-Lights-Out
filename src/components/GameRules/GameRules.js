import React from 'react';
import './GameRules.css';

const GameRules = () => {
  return (
    <div className="GameRules">
      <p>
        The game consists of a grid of lights which is 5x5 by default. When the
        game starts, a random number of these lights is switched on. The game
        default is a 50% chance of being lit. Pressing any of the lights will
        toggle it and the four adjacent lights. The goal of the puzzle is to
        switch all the lights off, preferably in as few button presses as
        possible.
      </p>
    </div>
  );
};

export default GameRules;
