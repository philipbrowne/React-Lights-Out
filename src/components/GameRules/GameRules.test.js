import React from 'react';
import { render } from '@testing-library/react';
import GameRules from './GameRules';

it('renders without crashing', () => {
  render(<GameRules />);
});

it('matches snapshot', () => {
  const { asFragment } = render(<GameRules />);
  expect(asFragment()).toMatchSnapshot();
});
