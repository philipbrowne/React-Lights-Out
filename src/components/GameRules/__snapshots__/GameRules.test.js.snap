// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="GameRules"
  >
    <p>
      The game consists of a grid of lights which is 5x5 by default. When the game starts, a random number or a stored pattern of these lights is switched on. Pressing any of the lights will toggle it and the four adjacent lights. The goal of the puzzle is to switch all the lights off, preferably in as few button presses as possible.
    </p>
  </div>
</DocumentFragment>
`;
